arrayobject/index.ts

```diff
@@ -12,6 +12,39 @@ export const getKeyValues = (data: Array<Record<string, unknown>>, key: string):
     return res;
 }
 
+export const countCharsInData = (data: string, chars: string[]): number => {
+    let counter: number = 0;
+    for (let i = 0; i < chars.length; i++) {
+        if(data.includes(chars[i])) {
+            counter++;
+        }
+    }
+    return counter;
+}
+
+export const sortObjectValuesByChars = (data: Array<Record<string, object>>, searchChars: string[]): Array<Record<string, object>> => {
+    let res:Array<Record<string, object>> = [];
+    let elementMatchCounter: number = 0;
+    data.forEach(element => {
+        let objMatchCounter: number = 0;
+        for (const key in element) {
+            const num: number = countCharsInData(element[key]+"" , searchChars)
```

Comment:
Refactor: before we loop in again, we could check if needle work is in the haystack e.g.. ade in adeola means searchChars.length we would save more time by bypassing the loop, if length is 0 loop comes in.
```suggestion
           if(key.includes(searchChars.join(","))){
           num = searchChars.length;
           } else {
           
            num: number = countCharsInData(element[key]+"" , searchChars)
            }
```